design/design-doc-custom-composition.md

```diff
@@ -0,0 +1,179 @@
+# Custom Composition Types
+
+* Owner: Muvaffak Onuş (@muvaf)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
```

Comment:
```suggestion
## Background
[Crossplane Composition](https://crossplane.github.io/docs/v0.12/introduction/composition.html) (with [associated refinements](https://github.com/crossplane/example-cnp)) supports authoring `CompositeResourceDefinitions` (XRDs) that publish CRDs to the Kubernetes API and `Composition` resources that provides a simple built-in YAML templating language for composing the cloud service primitives in Crossplane Providers for AWS, GCP, Azure, Alibaba, Packet, and more. You can define, compose, and offer new CRDs entirely in yaml without writing  any code, that can be consumed using `kubectl`, GitOps, or any tool that work with the Kubernetes API. This is suitable for many use cases and offers a Kubernetes-native authoring experience.

However in not everyone prefers authoring YAML, so having a toolkit like [cdk8s](https://github.com/awslabs/cdk8s) lets you compose cdk8s `Constructs` with your language of choice (Javascript, Python) with loops, conditionals, and a broad ecosystem of packages and libraries. A cdk8s `synth` command is run client-side, emits the yaml for a cdk8s program, and then `kubectl apply` can then be run to apply it to your cluster. 

cdk8s `Constructs` can't currently be published as native CRDs, so they can consumed via the Kubernetes API or composed into larger composition.
```

Reply:
Nit: The first paragraph here reads a bit like a sales pitch for composition, which isn't the tone we typically take in a design doc.

Perhaps we should try to incorporate this content while also making it clear that this type would be used for more than just CDK8s based compositions (e.g. for other external tools like Helm, Kustomize, etc) if that is indeed the case?